### PR TITLE
[types] Add chain_id and max_gas_amount to Transaction

### LINKED ISSUE #680: [Types] Add ChainID and GasLimit to transaction
ChainID: The transaction needs a ChainID field to distinguish the test network and the main network.
GasLimit: The transaction max used gas.
GasPricie: We do not need to add a GasPrice field #679 

#604

### PR DESCRIPTION
**Pull request by jolestar**
## Summary

1. Add  chain_id and max_gas_amount to Transaction.
2. Add sequence_number and max_gas_amount to TxContext.
3. Add chain_id as transaction_validator::validate argument.
4. Add TxResult to TxContext after tx is executed for the transaction validator to get gas_used in the post_execute function.

resolve #680

### CODE REVIEW

**Comment by jolestar** [comment]
cc @feliciss and @yubing744

File: crates/rooch-types/src/chain_id.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+use proptest_derive::Arbitrary;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+#[cfg_attr(any(test, feature = "fuzzing"), derive(Arbitrary))]
+#[repr(u64)]
+pub enum ChainID {
+    Dev = 20230103,
+    Test = 20230102,
+    Main = 20230101,
```
Comment (baichuan3) [inline crates/rooch-types/src/chain_id.rs:16]: Or directly represent by 1,2,3, it will be easier.
```
Dev = 3,
Test = 2,
Main = 1
```

Comment (jolestar) [reply crates/rooch-types/src/chain_id.rs:16]: Rooch's ChainID must not conflict with Ethereum or Ethereum fork chain id.

**Review by baichuan3 — APPROVED** [review]

**Comment by jolestar** [comment]
I merge this PR first. @steelgeek091 please help to resolve the conflict with #684

**Pull request merged**